File: backend/src/main/java/com/coing/domain/coin/market/dto/MarketResponseDto.java
package com.coing.domain.coin.market.dto;

import com.coing.domain.coin.market.entity.Market;

public record MarketResponseDto(
	String code,
	String koreanName,
	String englishName,
	Boolean isBookmarked
) {
	public static MarketResponseDto of(Market market, Boolean isBookmarked) {
		return new MarketResponseDto(
			market.getCode(),
			market.getKoreanName(),
			market.getEnglishName(),
			isBookmarked
		);
	}
}

File: backend/src/main/java/com/coing/domain/coin/market/dto/MarketResponseDto.kt
package com.coing.domain.coin.market.dto

import com.coing.domain.coin.market.entity.Market

data class MarketResponseDto(
	val code: String,
	val koreanName: String,
	val englishName: String,
	val isBookmarked: Boolean
) {
	companion object {
		fun of(market: Market, isBookmarked: Boolean): MarketResponseDto {
			return MarketResponseDto(
				code = market.code,
				koreanName = market.koreanName,
				englishName = market.englishName,
				isBookmarked = isBookmarked
			)
		}
	}
}